### ISSUE TITLE
Sample changing from class to interface breaks binary compatibilty

### ISSUE THREAD

**Issue by philwebb** [micrometer-metrics/micrometer#2006]
Commit 9fa83e3d3638edf725aa5d803ae0b52b9017456f changes `LongTaskTimer.Sample` from a class to an interface. This unfortunately breaks binary compatibility meaning all projects that use it will need to recompile.

```
java.lang.IncompatibleClassChangeError: Found interface io.micrometer.core.instrument.LongTaskTimer$Sample, but class was expected
	at org.springframework.batch.core.job.AbstractJob.execute(AbstractJob.java:366)
	at org.springframework.batch.core.launch.support.SimpleJobLauncher$1.run(SimpleJobLauncher.java:147)
	at org.springframework.core.task.SyncTaskExecutor.execute(SyncTaskExecutor.java:50)
	at org.springframework.batch.core.launch.support.SimpleJobLauncher.run(SimpleJobLauncher.java:140)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:198)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.batch.core.configuration.annotation.SimpleBatchConfiguration$PassthruAdvice.invoke(SimpleBatchConfiguration.java:127)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
```

**Comment by jkschneider** [comment]
@philwebb TIL that class -> interface breaks binary compatibility. Sorry for the trouble. Fixed in 75e6d9525ef1157d91d8bd19b4debdab79c0b60d

**Comment by philwebb** [comment]
I wouldn't have spotted it either. Thanks for the fast turnaround!
